docs/editor/testing.md

```diff
@@ -0,0 +1,173 @@
+- Use the commands to run tests from the Command Palette (`kb(workbench.action.showCommands)`)
+
+In the Test Explorer, use the controls in the section heading to run or debug all tests. You can also run or debug individual tests by selecting the play or debug icon next to the test name. In the tree view, when you select to run or debug on a specific node, all tests under that node are run or debugged.
+
+![Run and debug tests in Test Explorer](images/testing/run-debug-tests-test-explorer.png)
+
+While you're viewing your test code in the editor, you can use the play control in the gutter to run the test at the current cursor position. Right-click on the gutter control to view other actions, such as debugging the test.
+
+![Run and debug tests in editor gutter](images/testing/run-debug-tests-editor-gutter.png)
+
+> **Tip**: you can configure the default testing action for the gutter control by using the `testing.defaultGutterClickAction` setting.
+
+After running a test, the editor gutter displays the test status.
+
+When you run tests, the test result status is displayed in the Test Explorer view and editor gutter. The test results are color-coded to indicate the status of the test (pass or fail). When a test fails, notice that the test error message is shown as an overlay in the editor.
+
+Use the **Show Output** button in the Test Explorer to view the test output in the **Test Results** panel.
+
+![Test Results panel](images/testing/test-results-panel.png)
+
+You can access the commands for the Test Explorer, such as running or debugging tests, via the Command Palette (`kb(workbench.action.showCommands)`). Enter `Test Explorer:` to find the corresponding commands.
+
+## Test coverage
+
+Test coverage is a measure of how much of your code is covered by your tests. It helps you identify areas of your code that are not being tested. VS Code provides built-in support for running tests with coverage and viewing the coverage results.
+
+Similar to running and debugging tests, you can run tests with coverage by using the actions in the Test Explorer view, editor gutter, or commands in the Command Palette (`kb(workbench.action.showCommands)`).
+
+![Run tests with coverage](images/testing/run-tests-with-coverage.png)
+
+After you run tests with coverage, you can view the coverage results in different locations:
+
+- In the Test Coverage view
+
+    A tree view shows the tests with their coverage percentage. A color indicator also gives a visual cue about the coverage percentage. Hover over each line to see more details about the coverage results.
+
+    ![Test Coverage view](images/testing/test-coverage-view.png)
+
+- As an overlay in the editor
+
+    For code files that have test coverage, the editor shows a color overlay in the gutter to indicate which lines are covered by tests or not. When you hover over the gutter, notice that for covered lines, there is also an indicator for the number of times the line was executed.
```

Comment:
Should also mention command and the editor action (in the top right of the editor) to "toggle inline coverage" to show the code highlights persistently